src/workerd/util/perfetto-tracing-test.c++

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2025 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "perfetto-tracing.h"
+
+#include <kj/debug.h>
+#include <kj/test.h>
+
+#ifdef WORKERD_USE_PERFETTO
+#include "use-perfetto-categories.h"
+
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include <chrono>
+#include <thread>
+#endif  // WORKERD_USE_PERFETTO
+
+#include <cstdlib>
+
+namespace workerd {
+namespace {
+
+#ifdef WORKERD_USE_PERFETTO
+
+kj::String getTempFileName(const char* prefix) {
+  const char* tmpDir = getenv("TEST_TMPDIR");
+  if (tmpDir == nullptr) tmpDir = "/tmp";
+  return kj::str(tmpDir, "/", prefix, "-", rand(), ".pb");
+}
+
+bool fileExists(const char* path) {
+  struct stat buffer;
+  return (stat(path, &buffer) == 0);
+}
+
+size_t getFileSize(const char* path) {
+  struct stat buffer;
+  if (stat(path, &buffer) != 0) return 0;
+  return buffer.st_size;
+}
+
+void removeFile(const char* path) {
+  unlink(path);
+}
+
+KJ_TEST("PerfettoSession basic functionality") {
+  auto traceFile = getTempFileName("perfetto-test");
+
+  {
+    PerfettoSession session(traceFile, "workerd");
+
+    TRACE_EVENT("workerd", "test_event");
+    TRACE_EVENT("workerd", "test_event_with_args", "test_arg", 42);
+
+    TRACE_EVENT_BEGIN("workerd", "test_duration_event");
+    std::this_thread::sleep_for(std::chrono::microseconds(100));
+    TRACE_EVENT_END("workerd");
+
+    TRACE_EVENT_INSTANT("workerd", "test_instant_event");
+
+    TRACE_COUNTER("workerd", "test_counter", 100);
+    TRACE_COUNTER("workerd", "test_counter", 200);
+
+    session.flush();
+  }
+
+  KJ_ASSERT(fileExists(traceFile.cStr()));
+  KJ_ASSERT(getFileSize(traceFile.cStr()) > 0);
```

Comment:
This should likely test the contents of the file to confirm that the output looks somewhat correct based on the traced events.